### PR TITLE
Face identification module

### PR DESCRIPTION
**Pull request by Rob1in**

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	echo no sudo on this system, proceeding as current user
+	SUDO=""
+fi
+
+if command -v apt-get; then
+	if dpkg -l python3-venv; then
+		echo "python3-venv is installed, skipping setup"
+	else
+		if ! apt info python3-venv; then
+			echo package info not found, trying apt update
```
Comment (bhaney) [inline setup.sh]: specify that you were looking for the python3-env package

File: setup.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+			$SUDO apt-get -qq update
+		fi
+		$SUDO apt-get install -qqy python3-venv
```
Comment (bhaney) [inline setup.sh:20]: I'm not sure what would happen if viam-server was not running already running as root -- I don't think a user would get a prompt for their password

File: src/identifier.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    def compute_known_embeddings(self):
+        for label in self.labels_directories:
+            label_path = self.dataset_path+"/"+self.labels_directories[label]
+            embeddings = []
+            for file in os.listdir(label_path):
+                if (
+                    (".jpg" in file.lower())
+                    or (".jpeg" in file.lower())
+                    or (".png" in file.lower())
+                ):
```
Comment (bhaney) [inline src/identifier.py:70]: emit a warning in an else statement saying that the file did not end in ".jpeg", ".jpg", or ".png"

File: src/utils.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    Args:
+        raw_image (Union[Image.Image, RawImage])
+
+    Returns:
+        np.ndarray: BGR numpy array
+    """
+    if type(image) == RawImage:
+        im = Image.open(image.data).convert('RGB') #convert in RGB png openened in RGBA
+        return np.array(im)[...,::-1]
```
Comment (bhaney) [inline src/utils.py]: check the mime type here for acceptable mime types (which are just JPEG and PNG). In this way, you don't have to worry about custom Viam RGBA format, or the potential Viam Depth image

**Review by bhaney — CHANGES_REQUESTED** [review]
Nice work, some few changes, mostly around checking the types of images you are getting from the underlying camera

File: src/utils.py (hunk 4 above)
Comment (Rob1in) [reply src/utils.py]: done

File: src/identifier.py (hunk 3 above)
Comment (Rob1in) [reply src/identifier.py:70]: done

File: setup.sh (hunk 1 above)
Comment (Rob1in) [reply setup.sh]: done

File: setup.sh (hunk 2 above)
Comment (Rob1in) [reply setup.sh:20]: In my installation I get permission denied before reaching `setup.sh` so.....

Comment (bhaney) [reply setup.sh:20]: OK - looks you need to keep SUDO then

**Review by bhaney — APPROVED** [review]
LGTM!

**Pull request merged**